test/libsolidity/smtCheckerTests/abi/abi_encode_call_simple.sol

```diff
@@ -0,0 +1,26 @@
+contract C {
+	function callMeMaybe(uint a, uint b, uint[] memory c) external {}
+
+	function abiEncodeSimple(uint x, uint y, uint z, uint[] memory a, uint[] memory b) public view {
+		require(x == y);
+		bytes memory b1 = abi.encodeCall(this.callMeMaybe, (x, z, a));
+		bytes memory b2 = abi.encodeCall(this.callMeMaybe, (y, z, a));
+		assert(b1.length == b2.length);
+
+		bytes memory b3 = abi.encodeCall(this.callMeMaybe, (y, z, b));
+		assert(b1.length == b3.length); // should fail
+	}
+}
+// ====
+// SMTEngine: all
+// SMTIgnoreCex: yes
+// ----
+// Warning 6031: (233-249): Internal error: Expression undefined for SMT solver.
+// Warning 6031: (298-314): Internal error: Expression undefined for SMT solver.
+// Warning 6031: (398-414): Internal error: Expression undefined for SMT solver.
```

Comment:
That's not nice... I've been working on removing this kind of stuff, so it would be nice if this was fixed before merging.
I think this has to do with shortcutting `bool SMTEncoder::visit(FunctionCall const& _funCall)`, like it does with `ABIDecode`.

Reply:
Actually no, now I think this is because of my other comment, since it's trying to get the smt expression of the tuple argument (still need to investigate more)